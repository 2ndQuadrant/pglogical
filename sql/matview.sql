/* First test whether a table's replication set can be properly manipulated */
SELECT * FROM pglogical_regress_variables()
\gset

\c :provider_dsn

SELECT pglogical.replicate_ddl_command($$
CREATE TABLE public.test_tbl(id serial primary key, data text);
CREATE MATERIALIZED VIEW public.test_mv AS (SELECT * FROM public.test_tbl);
$$);

SELECT * FROM pglogical.replication_set_add_all_tables('default', '{public}');

INSERT INTO test_tbl VALUES (1, 'a');

REFRESH MATERIALIZED VIEW test_mv;

INSERT INTO test_tbl VALUES (2, 'b');

SELECT pg_xlog_wait_remote_apply(pg_current_xlog_location(), 0);

SELECT * FROM test_tbl;
SELECT * FROM test_mv;

\c :subscriber_dsn

SELECT * FROM test_tbl;
SELECT * FROM test_mv;

\c :provider_dsn
\set VERBOSITY terse
SELECT pglogical.replicate_ddl_command($$
	DROP TABLE public.test_tbl CASCADE;
$$);
