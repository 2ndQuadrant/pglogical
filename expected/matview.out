/* First test whether a table's replication set can be properly manipulated */
SELECT * FROM pglogical_regress_variables()
\gset
\c :provider_dsn
SELECT pglogical.replicate_ddl_command($$
CREATE TABLE public.test_tbl(id serial primary key, data text);
CREATE MATERIALIZED VIEW public.test_mv AS (SELECT * FROM public.test_tbl);
$$);
 replicate_ddl_command 
-----------------------
 t
(1 row)

SELECT * FROM pglogical.replication_set_add_all_tables('default', '{public}');
 replication_set_add_all_tables 
--------------------------------
 t
(1 row)

INSERT INTO test_tbl VALUES (1, 'a');
REFRESH MATERIALIZED VIEW test_mv;
INSERT INTO test_tbl VALUES (2, 'b');
SELECT pg_xlog_wait_remote_apply(pg_current_xlog_location(), 0);
 pg_xlog_wait_remote_apply 
---------------------------
 
(1 row)

SELECT * FROM test_tbl;
 id | data 
----+------
  1 | a
  2 | b
(2 rows)

SELECT * FROM test_mv;
 id | data 
----+------
  1 | a
(1 row)

\c :subscriber_dsn
SELECT * FROM test_tbl;
 id | data 
----+------
  1 | a
  2 | b
(2 rows)

SELECT * FROM test_mv;
 id | data 
----+------
(0 rows)

\c :provider_dsn
\set VERBOSITY terse
SELECT pglogical.replicate_ddl_command($$
	DROP TABLE public.test_tbl CASCADE;
$$);
NOTICE:  drop cascades to materialized view public.test_mv
NOTICE:  drop cascades to table public.test_tbl membership in replication set default
 replicate_ddl_command 
-----------------------
 t
(1 row)

